Add better writeup for wasm (#1458)
stellar/stellar-docs: docs/learn/fundamentals/stellar-data-structures/contracts.mdx

Once a smart contract has been written by a developer and compiled into a Wasm executable file, it can then be "installed" onto the Stellar network. A `CONTRACT_DATA` [ledger entry](./ledgers.mdx) is created to store this binary data and its unique identifier is the hash of the executable file. This binary executable is stored independently from its deployed contract(s). When a Stellar transaction attempts to invoke a contract function, the Wasm bytecode is first retrieved from the ledger and a secure, isolated runtime virtual machine ("VM") is instantiated so it can run the bytecode for the contract and then exit.

### WebAssembly (Wasm) in Stellar Smart Contracts

WebAssembly (Wasm) is a lightweight, portable binary instruction format designed for high-performance execution across various environments, including blockchains, web browsers, and cloud services. In Stellar, Wasm serves as the foundation for smart contracts by enabling a secure and efficient execution environment.

### Wasm Compilation and Installation

> Once a developer writes a smart contract in a supported language, such as Rust, the contract is compiled into a Wasm executable file. This file contains low-level bytecode that can be efficiently executed within a sandboxed environment. After compilation, the Wasm executable is "installed" on the Stellar network, where it is stored as a `CONTRACT_DATA` [ledger entry](./ledgers.mdx) . The unique identifier for this entry is derived from the hash of the compiled binary and is called `Wasm hash`.

Reviewer: ```suggestion
Once a developer writes a smart contract in a supported language, such as Rust, the contract is compiled into a Wasm executable file. This file contains low-level bytecode that can be efficiently executed within a sandboxed environment. After compilation, the Wasm executable is uploaded to the Stellar network, where it is stored as a `CONTRACT_DATA` [ledger entry](./ledgers.mdx) . The unique identifier for this entry is derived from the hash of the compiled binary and is called `Wasm hash`.
```

Author: Updated

> To retrieve Wasm executable file from the Stellar network to "install" in your local network, it is possible to do so using the RPC's `getLedgerEntries` or use [Stellar Lab's Contract Explorer](https://lab.stellar.org/smart-contracts/contract-explorer).

Reviewer: ```suggestion
To retrieve Wasm executable file from the Stellar network to upload to your local network, it is possible to do so using the RPC's `getLedgerEntries` or use [Stellar Lab's Contract Explorer](https://lab.stellar.org/smart-contracts/contract-explorer).
```
